### DISCUSSION TITLE
Improve tests speed

### DISCUSSION THREAD

**Discussion by IvanovRoman** [cypress-io/cypress#28391] [Questions and Help]
I use component testing for my tests and use webpack to build code. The wait between tests runs is very long. When I enabled `DEBUG=cypress:*` mode, I noticed that these lines add 2 seconds to the time between tests.
```
cypress:proxy:http:util:buffers resetting buffers +1s
cypress:server:automation:cookies getting:cookies {} +1s
```
How can I disable them to speed up my tests?

**Comment by portyu9** [comment]
Hi @IvanovRoman 

Those `DEBUG` lines do not necessarily mean that those two operations themselves each consume a fixed one-second delay.

The `+1s` value in Cypress's debug output is the elapsed time since the previous message in that debug namespace, so I would not try to disable `resetting buffers` or `getting:cookies` based on those timestamps alone. They are Cypress internals, not supported performance switches.

For Component Testing, first measure where the delay occurs:

- between individual `it()` tests;
- when a new spec is opened;
- while Webpack recompiles;
- while the browser/Cypress Runner is doing work.

If most of the delay is **between spec files in `cypress run`**, current Cypress has:

```js
const { defineConfig } = require('cypress')

module.exports = defineConfig({
  component: {
    experimentalSingleTabRunMode: true,
  },
})
```

This runs component specs in a single browser tab and can improve run-mode performance.

It is still experimental, and Cypress warns that it can affect isolation/reliability on large suites, so compare the suite with and without it.

If the delay is during Webpack rebuilds instead, profile the Component Testing dev-server configuration and make sure you are not compiling unnecessary application files/plugins/loaders for each spec.

You can also temporarily disable the Cypress Command Log to determine whether runner rendering contributes significantly:

```bash
CYPRESS_NO_COMMAND_LOG=1 npx cypress run --component
```

That is useful as a diagnostic, not something I would immediately make the permanent solution.

Please mark this answer as accepted if it helped, thank you.
